src/internal/snapshotdb/snapshot_crud_test.go

```diff
@@ -0,0 +1,75 @@
+package snapshotdb
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dbutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	snapshotpb "github.com/pachyderm/pachyderm/v2/src/snapshot"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+func TestCreateAndGetJob(t *testing.T) {
+	withDependencies(t, func(d dependencies) {
+		metadata := map[string]string{"key": "value"}
+		id, err := CreateSnapshot(d.ctx, d.tx, d.s, metadata)
+		if err != nil {
+			t.Fatalf("create snapshot in database: %v", err)
+		}
+		snpshot, err := GetSnapshot(d.ctx, d.tx, id)
+		if err != nil {
+			t.Fatalf("get snapshot %d from database: %v", id, err)
+		}
+		t.Log(snpshot)
+	})
+}
+
+func TestListSnapshotTxByFilter(t *testing.T) {
+	ctx, db := DB(t)
+	fs := FilesetStorage(t, db)
+	want := make([]*snapshotpb.SnapshotInfo, 0)
+	err := dbutil.WithTx(ctx, db, func(ctx context.Context, sqlTx *pachsql.Tx) error {
+		_, err := CreateSnapshot(ctx, sqlTx, fs, map[string]string{})
+		if err != nil {
+			t.Fatalf("create snapshot in iteration 1: %v", err)
```

Comment:
Don't fail the test inside a transaction, just return an error.  The transaction can still be re-run at this point; the error from WithTx is the correct place to fail the test.